### ISSUE TITLE
How to deal with overlapping labels?

### ISSUE THREAD

**Issue by Cocomolch4000** [stardist/stardist#140]
Hello,

I took a look at the MoNuSeg 2018 training data for HE images.
There, the GT labels overlap. How would I pass stardist overlapping GT labels?
So, how do I define X, Y for training if some labels in Y overlap and are not unique.

Best
Cocomolch

**Comment by maweigert** [comment]
Hi @Cocomolch4000 ,

unfortunately it is currently not possible to use overlapping labels - so you would need to create single label masks from the multi channel MoNuSeg masks (e.g. by max projecting). 

Cheers,

Martin

**Comment by Cocomolch4000** [comment]
Hi Martin,

Thanks for the quick reply!
Since the pretrained HE model uses this dataset, did you generate single label masks during training?
And in doing so wouldn't one drop crucial neighborhood information?
Would you see hindrances to just cutting labels where they overlap? There would be obviously an information loss, however, one would incorporate neighborhood information during training?

Sorry for so many questions!
Best
Cocomolch

**Comment by maweigert** [comment]
> Since the pretrained HE model uses this dataset, did you generate single label masks during training?

Yes, we simply max projected the labels. 

> And in doing so wouldn't one drop crucial neighborhood information?

What do you mean by that?

> Would you see hindrances to just cutting labels where they overlap? There would be obviously an information loss, however, one would incorporate neighborhood information during training?

Yes, one could as well just remove the overlapping areas. We however didn't really look into the effects of doing so systematically yet.

**Comment by Cocomolch4000** [comment]
Do I understand max projection correctly: for every single nucleus you have a single label mask resulting in many labelmasks Y_1 ... Y_N for only one input image X? If thats right - in order to lower overall cost - the network still will give all segmentations for the image in the best case. However, wouldn't it be helpful to the network to learn the constraining neighbors of each nucleus at a time? Or is this implicitly also captured if one has many label matrices corresponding to one image?

**Comment by maweigert** [comment]
In the MoNuSeg paper (Kumar et al 2019), the authors do the same 

`"For overlapping nuclei, we assigned each multi-nuclear pixel to the largest nucleus containing that pixel."`

So I think removing the overlaps is fine.

**Comment by Cocomolch4000** [comment]
Alright, thanks!
